canary/producer-c/canary/KvsProducerSampleCloudwatch.cpp

```diff
@@ -251,8 +252,11 @@ STATUS initWithEnvVars(PCanaryConfig pCanaryConfig)
 
     pCanaryConfig->bufferDuration = pCanaryConfig->bufferDuration * HUNDREDS_OF_NANOS_IN_A_SECOND;
 
-    if (pCanaryConfig->useIotCredentialProvider) {
+    if (pCanaryConfig->useIotCredentialProvider == TRUE) {
         CHK_STATUS(mustenv(IOT_CORE_CREDENTIAL_ENDPOINT_ENV_VAR, pCanaryConfig->iotCoreCredentialEndPoint));
+        CHK_STATUS(readFile(pCanaryConfig->iotCoreCredentialEndPoint, TRUE, NULL, &fileSize));
+        CHK_STATUS(readFile(pCanaryConfig->iotCoreCredentialEndPoint, TRUE,pCanaryConfig->iotEndpoint, &fileSize));
+        pCanaryConfig->iotEndpoint[fileSize - 1] = '\0';
```

Comment:
If fileSize is greater than 1023 and the buffer cannot accommodate, readFile call fails anyways. So we are covered there. If fileSize is 0, the IoT calls would fail, but now that I think about it, we should exit with an application status code so that the cause of failure is clear.